compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -427,12 +427,13 @@ static LogicalResult setRootConfigForPackPeel4LevelTilingPipeline(
     unpackEmpty = {false, false, true};
     innerPerm = {innerPermA, innerPermB, {0, 1}};
     outerPerm = {outerPermA, outerPermB};
-    // Add outer permutation for unpack. NOTE: This currently fails for some
-    // tests in the AIR pipeline.
+    // Add outer permutation for unpack.
     if (isBatchMatmul) {
       outerPerm.push_back({0, 2, 1});
     } else {
-      outerPerm.push_back({1, 0});
+      if (isObjectFifo) {
+        outerPerm.push_back({1, 0});
+      }
```

Comment:
So for AIR the outerPerm needs to be {0, 1}? Does it work if you don't push anything?

Reply:
The title is a bit confusing... Are you using the same outerPerm or not?

Reply:
The matmul test on the CI seems to be passing for both with and without the outerPerm... I haven't been able to investigate deeply why both configurations would pass the same test. This PR was intended to make it follow the same code structure as the pack-peel pipeline.

Reply:
Make sense, will definitely start sweeping the GEMM sizes and will keep the outerPerm consideration in mind! I will close the PR for now, and follow up afterwards.